src/migrations/20220425000000-goal-templates.js

```diff
@@ -0,0 +1,1964 @@
+              tableName: 'Roles',
+            },
+            key: 'id',
+          },
+          onUpdate: 'CASCADE',
+          onDelete: 'CASCADE',
+        },
+        createdAt: {
+          allowNull: false,
+          type: Sequelize.DATE,
+        },
+        updatedAt: {
+          allowNull: false,
+          type: Sequelize.DATE,
+        },
+      }, { transaction });
+
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // Disable logging while doing mass updates
+      try {
+        await queryInterface.sequelize.query(
+          `
+          SELECT "ZAFSetTriggerState"(null, null, null, 'DISABLE');
+          `,
+          { transaction },
+        );
+      } catch (err) {
+        console.error(err); // eslint-disable-line no-console
+        throw (err);
+      }
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+      // -------------------------------------------------------------------------------------------
+
+      // Clean up Roles table and add isSpecialist, deletedAt, & mapsTo columns
+      try {
+        await queryInterface.addColumn('Roles', 'isSpecialist', {
+          type: Sequelize.BOOLEAN,
+          allowNull: true,
+          default: false,
+          onUpdate: 'CASCADE',
+        }, { transaction });
+
+        await queryInterface.addColumn('Roles', 'deletedAt', {
+          type: Sequelize.DATE,
+          allowNull: true,
+        }, { transaction });
+
+        await queryInterface.addColumn('Roles', 'mapsTo', {
+          type: Sequelize.INTEGER,
+          allowNull: true,
+        }, { transaction });
+
+        await queryInterface.sequelize.query(
+          `DO $$
+          BEGIN
+            CREATE TEMP TABLE "TempRoles" AS
+            SELECT
+              MIN(r."id") "id",
+              r.name,
+              r."fullName",
+              MIN(r."createdAt") "createdAt",
+              MAX(r."updatedAt") "updatedAt",
+              MAX(r."updatedAt") "deletedAt",
```

Comment:
Is this filling the `deletedAt` field in the new table as the max `updatedAt`? Wouldn't we want the new table to have a null `deletedAt` by default?